api/tests/routes/auth/discord_test.py

```diff
@@ -0,0 +1,243 @@
+        )
+        response = self.post_to_endpoint(client, form=form_data)
+        assert response.status_code == 200
+
+    def test_account_deleted_account_state(self, client):
+        redirect_url_scheme = "https"
+        redirect_url_netloc = "test.com"
+        form_data = {
+            "email": "user@test.com",
+            "password": settings.TEST_DEFAULT_PASSWORD,
+            "discord_id": "1234",
+            "redirect_url": f"{redirect_url_scheme}://{redirect_url_netloc}",
+        }
+        user = users_factories.UserFactory(email=form_data["email"], password=form_data["password"], isActive=False)
+        history_factories.SuspendedUserActionHistoryFactory(user=user, reason=users_constants.SuspensionReason.DELETED)
+
+        response = self.post_to_endpoint(client, form=form_data)
+        assert response.status_code == 200
+
+    def test_allow_inactive_user_sign(self, client):
+        redirect_url_scheme = "https"
+        redirect_url_netloc = "test.com"
+        form_data = {
+            "email": "user@test.com",
+            "password": settings.TEST_DEFAULT_PASSWORD,
+            "discord_id": "1234",
+            "redirect_url": f"{redirect_url_scheme}://{redirect_url_netloc}",
+        }
+        users_factories.AnonymizedUserFactory(email=form_data["email"], password=form_data["password"], isActive=False)
+        response = self.post_to_endpoint(client, form=form_data)
+        assert response.status_code == 200
+
+    def test_unknown_user_logs_in(self, client, caplog):
+        redirect_url_scheme = "https"
+        redirect_url_netloc = "test.com"
+        form_data = {
+            "email": "user@test.com",
+            "password": settings.TEST_DEFAULT_PASSWORD,
+            "discord_id": "1234",
+            "redirect_url": f"{redirect_url_scheme}://{redirect_url_netloc}",
+        }
+        response = self.post_to_endpoint(client, form=form_data)
+        assert response.status_code == 200
+
+    def test_user_without_password_logs_in(self, client, caplog):
+        user = users_factories.UserFactory(password=None, isActive=True)
+        redirect_url_scheme = "https"
+        redirect_url_netloc = "test.com"
+        form_data = {
+            "email": user.email,
+            "password": settings.TEST_DEFAULT_PASSWORD,
+            "discord_id": "1234",
+            "redirect_url": f"{redirect_url_scheme}://{redirect_url_netloc}",
+        }
+        response = self.post_to_endpoint(client, form=form_data)
+        assert response.status_code == 200
+
+    def test_user_logs_in_with_missing_fields(self, client):
+        redirect_url_scheme = "https"
+        redirect_url_netloc = "test.com"
+        form_data = {
+            "email": "user@test.com",
+            "discord_id": "1234",
+            "redirect_url": f"{redirect_url_scheme}://{redirect_url_netloc}",
+        }
+        response = self.post_to_endpoint(client, form=form_data)
+        assert response.status_code == 200
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏On vaut vraiment des 200 pour tous ces cas limites ? ça me parait pas clair :/

Reply:
oui c'est des cas d'erreur dans ce cas on affiche la page de la route get avec l erreur en plus